examples/particles.c

```diff
@@ -457,7 +457,12 @@ static void draw_particles(GLFWwindow* window, double t, float dt)
     {
         struct timespec ts;
         clock_gettime(CLOCK_REALTIME, &ts);
-        ts.tv_nsec += 100000000;
+        ts.tv_nsec += 100 * 1000 * 1000;
+        if (ts.tv_nsec >= 1000 * 1000 * 1000)
```

Comment:
It should be `if`, so the patch is valid.

`if` because `clock_gettime()` is guaranteed to write a valid value for `ts` (such that `ts.tv_nsec >= 0 && tv_nsec < 1e9`), at least when `clock_gettime()` succeeds (returns 0), and because the added value (= timeout) to `ts.tv_nsec` is only 100e6.
Thus in the worst case, `ts.tv_nsec` can become 1_099_999_999 (= (1e9 - 1) + 100e6), which is smaller than 2e9, thus at most one subtraction needs to be done, hence no `while` loop needed.

Note that when `clock_gettime()` doesn't succeed, which could be possible when `CLOCK_REALTIME` is not valid for the OS (and thus when it would return `EINVAL`), `ts` is *not* valid anymore.
Actually, this should be checked, but considering almost no error checking is done for the rest of that example (e.g., see the call to `cnd_timedwait()`; it's only an example after all), I did not change that part.